crates/nu-plugin/src/protocol/plugin_custom_value.rs

```diff
@@ -22,41 +22,171 @@ mod tests;
 /// values sent matches the plugin it is being sent to.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct PluginCustomValue {
-    /// The name of the custom value as defined by the plugin (`value_string()`)
-    pub name: String,
-    /// The bincoded representation of the custom value on the plugin side
-    pub data: Vec<u8>,
+    #[serde(flatten)]
+    shared: Arc<SharedContent>,
```

Comment:
Yeah, exactly. I documented that behavior. But the idea basically is that the plugin should be able to be notified when all of the copies of the custom value go out of scope - this is for handles and such.

The alternative to using an `Arc` here would be to notify the plugin on both `clone_value` and on drop, so that it can handle increment/decrement properly, but I didn't really love that idea. Instead it's relatively simple - if you send a marked custom value to the engine, you'll get notified when all copies of it go out of scope, but if you get sent it back and then send it again that counts as a new instance so you'll have to handle that separately.

If we really don't want to enable the `rc` feature of serde, I can do a manual impl of `Serialize` and `Deserialize` on a wrapper - that should be fairly trivial. I completely understand not wanting to have it

Reply:
That makes sense as the GC we have at home.

I think I am fine with `rc` for now. I probably need to read up a bit more about your architecture, because I am unsure what is happening on the plugin side and what is happening on the Nushell side.

Reply:
It's possible that plugins themselves might not be happy with the `rc` feature being set, so I'm thinking I might just do this with a custom impl anyway. It shouldn't add very much code to do it

Reply:
This is done now. `rc` feature no longer needed.

Reply:
Sounds good to me.